src/components/tl/mlx5/tl_mlx5.h

```diff
@@ -110,13 +110,14 @@ typedef enum
 
 typedef struct ucc_tl_mlx5_team {
     ucc_tl_team_t             super;
-    ucc_status_t              status[2];
     ucc_service_coll_req_t   *scoll_req;
     ucc_tl_mlx5_team_state_t  state;
+    ucc_status_t              dm_status[2];
```

Comment:
Can you please avoid using hard-coded values as much as possible? Use an enum to refer to each device's memory status, for example, 
dm_status[0]  --> dm_status[LOCAL_DM_STATUS]
dm_status[1]  --> dm_status[GLOBAL_DM_STATUS]
 ucc_status_t   status[2] -->  ucc_status_t status[DM_STATUS_ARRAY_SIZE]

Reply:
I agree with you that this was not clean. I defined a struct in `tl_mlx5.h` to handle this:
```
typedef struct ucc_tl_mlx5_team_status {
    ucc_status_t local;
    ucc_status_t global;
} ucc_tl_mlx5_team_status_t;
```
Lmk if it is good for you.
We might reuse this struct down the line when adding more collectives, to precisely gather and store the status of each component of TL/mlx5.